db/db_impl_write.cc

```diff
@@ -534,23 +566,76 @@ Status DBImpl::PipelinedWriteImpl(const WriteOptions& write_options,
   return w.FinalStatus();
 }
 
+Status DBImpl::UnorderedWriteMemtable(const WriteOptions& write_options,
+                                      WriteBatch* my_batch,
+                                      WriteCallback* callback, uint64_t log_ref,
+                                      SequenceNumber seq) {
+  PERF_TIMER_GUARD(write_pre_and_post_process_time);
+  StopWatch write_sw(env_, immutable_db_options_.statistics.get(), DB_WRITE);
+
+  WriteThread::Writer w(write_options, my_batch, callback, log_ref,
+                        false /*disable_memtable*/);
+
+  size_t total_count = 0;
+  size_t total_byte_size = 0;
+
+  if (w.CheckCallback(this)) {
+    if (w.ShouldWriteToMemtable()) {
+      w.sequence = seq;
+      total_count += WriteBatchInternal::Count(my_batch);
+    }
+    total_byte_size = WriteBatchInternal::AppendedByteSize(
+        total_byte_size, WriteBatchInternal::ByteSize(w.batch));
+  }
+
+  auto stats = default_cf_internal_stats_;
+  stats->AddDBStats(InternalStats::NUMBER_KEYS_WRITTEN, total_count);
+  RecordTick(stats_, NUMBER_KEYS_WRITTEN, total_count);
+  stats->AddDBStats(InternalStats::BYTES_WRITTEN, total_byte_size);
+  RecordTick(stats_, BYTES_WRITTEN, total_byte_size);
+  RecordInHistogram(stats_, BYTES_PER_WRITE, total_byte_size);
+
+  if (!w.CallbackFailed()) {
+    WriteStatusCheck(w.status);
+  }
+
+  ColumnFamilyMemTablesImpl column_family_memtables(
+      versions_->GetColumnFamilySet());
+  w.status = WriteBatchInternal::InsertInto(
+      &w, w.sequence, &column_family_memtables, &flush_scheduler_,
+      write_options.ignore_missing_column_families, 0 /*log_number*/, this,
+      true /*concurrent_memtable_writes*/);
+  auto pending_cnt = pending_memtable_writes_.fetch_sub(1) - 1;
```

Comment:
`auto` here doesn't help readability. I suggest we spell out the type.

Reply:
Sure, will do.